### PR TITLE
[11.0][MIG] mrp_production_request

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Migration to v11.

To Do:
* [x] : Propagation to/from stock.moves.
* [x] : Improve coverage?
* [x] : Reduce qty needed to procure when MRs are open.
* [x] : Button to access MOs.
* [x] : Migration scripts
* [x] : Update README

### CODE REVIEW

**Comment by HviorForgeFlow** [comment]
Functional test:
Administrator as a Manufacturing Request Manager
Demo user as Manufacturing Request User + Inventory User + Manufacturing User
- If a Manufacturing Request User is not in Manufacturing User group at least can't access to use Manufacturing Requests.

Once you reach Manufacturing Request and create a record, you can assign anybody to be approved, could be resolved using [this approach]( https://github.com/OCA/purchase-workflow/blob/5c4dfab9f3f2ecd8b64a044de45f5d9f637519fd/purchase_request/models/purchase_request.py#L74)

Anyway, I can continue creating a Manufacture Request. But when I'm ready to save it. 
 
image

File: mrp_production_request/tests/test_mrp_production_request.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        self.bom_model = self.env['mrp.bom']
+        self.group_model = self.env['procurement.group']
+        self.product_model = self.env['product.product']
+        self.bom_model = self.env['mrp.bom']
+        self.boml_model = self.env['mrp.bom.line']
+
+        self.warehouse = self.env.ref('stock.warehouse0')
+        self.stock_loc = self.env.ref('stock.stock_location_stock')
+        route_manuf = self.env.ref('mrp.route_warehouse0_manufacture')
+        # route_mto = self.env.ref('stock.route_warehouse0_mto')
```
Comment (HviorForgeFlow) [inline mrp_production_request/tests/test_mrp_production_request.py]: Delete unused lines.

**Comment by LoisRForgeFlow** [comment]
@hveficent Thanks for noticing, that error happens in estandard Odoo with standard MOs, I have reported and issue: https://github.com/odoo/odoo/issues/28467.

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

File: mrp_production_request/models/procurement_rule.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ProcurementOrder(models.Model):
```
Comment (HviorForgeFlow) [inline mrp_production_request/models/procurement_rule.py:8]:
```suggestion
class ProcurementRule(models.Model):
```

**Review by HviorForgeFlow — APPROVED** [review]
:+1: LGTM just a minor change.

**Comment by JordiBForgeFlow** [comment]
@lreficent cam you squash the last commit? This one is ready to merge

**Comment by LoisRForgeFlow** [comment]
Rebase done and green!

**Pull request merged**